service/src/main/java/greencity/service/UserServiceImpl.java

```diff
@@ -112,13 +113,16 @@ public UserStatusDto updateStatus(Long id, UserStatus userStatus, String email)
     /**
      * {@inheritDoc}
      */
+    @Deprecated
     @Override
+    @Transactional
     public UserRoleDto updateRole(Long id, Role role, String email) {
         checkUpdatableUser(id, email);
-        UserVO userVO = findById(id);
-        userVO.setRole(role);
-        userRepo.updateUserRole(id, String.valueOf(role));
-        return modelMapper.map(userVO, UserRoleDto.class);
+        User user = userRepo.findById(id)
+            .orElseThrow(() -> new WrongIdException(ErrorMessage.USER_NOT_FOUND_BY_ID + id));
+        user.setRole(role);
+        userRepo.save(user);
+        return modelMapper.map(user, UserRoleDto.class);
     }
```

Comment:
Sonsarcube suggests adding @deprecated also to the javadocs, in order to explain what is to be used instaead for other developers.